src/main/java/org/example/keycloak/UserApiService.java

```diff
@@ -30,10 +30,10 @@ public List<User> getAllUsers(String url){
 
             if (responseCode == HttpURLConnection.HTTP_OK) {
                 BufferedReader reader = new BufferedReader(new InputStreamReader(connection.getInputStream()));
-                String allUsersFromApi = reader.readLine();
+                String allUsersFromApi ;
                 StringBuilder response = new StringBuilder();
 
-                while (Objects.nonNull(allUsersFromApi)) {
```

Comment:
here this variable was not updating in the loop. That is why I have to instantiate it in while loop only. Before it was giving me error regarding heap memory Out of storage error.